### PR TITLE
opentelemetry-instrumentation-openai-v2: scrub cookie from tests

### PR DESCRIPTION
**Pull request by codefromthecrypt**
# Description

This scrubs test data and rewords token replacements corresponding to canonical ENV variables.

Before, we had inconsistent test data where most sensitive properties were scrubbed, but not all. The main thing this does is scrub out the request cookie, so that folks don't accidentally leak it in new PRs. After that, it helps clarify the source of sensitive data by making the replacement tokens match naming conventions of OpenAI ENV variables.

Also, before we scrubbed "api-key", but that is only set when using `AzureClient`, which this package doesn't use, yet. When it does, we should probably scrub more (as the URLs include sensitive project-specific details). If we end up supporting `AzureClient`, I can help in the scrub config for that.

Noticed the cookie problem in #2984
See https://github.com/openai/openai-python/blob/646a579cdb305a9d3fba6c5f9a96011c5e2c2882/src/openai/_client.py#L98-L100

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I deleted all test yaml and ran the test two ways to make sure it recreates correctly, with no cookies in the yaml
- [x] `tox -e py312-test-instrumentation-openai-v2-0`
- [x] `tox -e py312-test-instrumentation-openai-v2-1`

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: instrumentation-genai/opentelemetry-instrumentation-openai-v2/tests/cassettes/test_chat_completion_tool_calls_no_content.yaml [hunk 1]
```diff
@@ -218,19 +226,20 @@ interactions:
...
       - gzip, deflate
+      authorization:
+      - Bearer test_openai_api_key
       connection:
       - keep-alive
       content-length:
       - '746'
       content-type:
       - application/json
       cookie:
```
Comment (codefromthecrypt) [inline instrumentation-genai/opentelemetry-instrumentation-openai-v2/tests/cassettes/test_chat_completion_tool_calls_no_content.yaml:237]: here's the intended and most important impact

File: instrumentation-genai/opentelemetry-instrumentation-openai-v2/tests/conftest.py [hunk 2]
```diff
@@ -66,7 +66,12 @@ def openai_client():
 @pytest.fixture(scope="module")
 def vcr_config():
     return {
-        "filter_headers": ["authorization", "api-key"],
+        "filter_headers": [
+            ("cookie", "test_cookie"),
```
Comment (codefromthecrypt) [inline instrumentation-genai/opentelemetry-instrumentation-openai-v2/tests/conftest.py:70]: here's the replacement and the way to add substitutions on request side (similar to how we do in response side)

**Review by codefromthecrypt — COMMENTED** [review]
notes

**Comment by alizenhom** [comment]
lgtm @codefromthecrypt

**Review by alizenhom — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
Thanks @alizenhom can you put the skip changelog label on for me? I don't have karma for that

**Review by karthikscale3 — APPROVED** [review]

**Review by xrmx — APPROVED** [review]
Good catch!

**Pull request merged**